### ISSUE TITLE
Training too slow

### ISSUE THREAD

**Issue by maskrcnnuser** [dbolya/yolact#517]
@dbolya - I have put the following comment in a closed issue #43 Not sure if it will be picked up or not so creating a new issue.

I am training using my set of 20 images for training and 10 more images for validation - I purposefully kept the set small in the beginning to test out the system. However, my output says ETA 58 days!! My config is as follows.

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 426.23 Driver Version: 426.23 CUDA Version: 10.1 |
|-------------------------------+----------------------+----------------------+
| GPU Name TCC/WDDM | Bus-Id Disp.A | Volatile Uncorr. ECC |
| Fan Temp Perf Pwr:Usage/Cap| Memory-Usage | GPU-Util Compute M. |
|===============================+======================+======================|
| 0 Tesla K80 TCC | 00000001:00:00.0 Off | 0 |
| N/A 80C P0 143W / 149W | 10404MiB / 11448MiB | 96% Default |
+-------------------------------+----------------------+----------------------+

Timer varies between 8.5 and 11.6 seconds.

Partial output is as follows:

[2490]    4980 || B: 0.242 | C: 0.001 | M: 0.286 | S: 0.020 | T: 0.549 || ETA: 58 days, 8:07:52 || timer: 8.641

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.10 |  0.29 |  0.24 |  0.24 |  0.10 |  0.10 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.12 |  0.56 |  0.21 |  0.13 |  0.10 |  0.10 |  0.10 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.09 |  0.30 |  0.29 |  0.11 |  0.09 |  0.09 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.17 |  0.55 |  0.55 |  0.54 |  0.09 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.09 |  0.29 |  0.23 |  0.11 |  0.11 |  0.11 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.22 |  0.65 |  0.65 |  0.65 |  0.11 |  0.11 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

[2495]    4990 || B: 0.244 | C: 0.001 | M: 0.287 | S: 0.020 | T: 0.552 || ETA: 58 days, 7:55:21 || timer: 11.297

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.08 |  0.24 |  0.22 |  0.22 |  0.13 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.16 |  0.76 |  0.29 |  0.17 |  0.13 |  0.13 |  0.13 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.09 |  0.27 |  0.24 |  0.22 |  0.13 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.22 |  0.66 |  0.65 |  0.65 |  0.13 |  0.13 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

[2500]    5000 || B: 0.246 | C: 0.001 | M: 0.286 | S: 0.020 | T: 0.553 || ETA: 58 days, 7:36:54 || timer: 8.562

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.08 |  0.25 |  0.22 |  0.20 |  0.11 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.20 |  0.80 |  0.79 |  0.16 |  0.16 |  0.11 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.07 |  0.19 |  0.16 |  0.16 |  0.09 |  0.09 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.09 |  0.59 |  0.09 |  0.09 |  0.09 |  0.09 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.07 |  0.19 |  0.16 |  0.16 |  0.09 |  0.09 |  0.01 |  0.01 |  0.00 |  0.00 |  0.00 |
  mask |  0.08 |  0.55 |  0.14 |  0.07 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

[2505]    5010 || B: 0.243 | C: 0.001 | M: 0.282 | S: 0.020 | T: 0.547 || ETA: 58 days, 7:22:32 || timer: 11.594

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.06 |  0.16 |  0.14 |  0.14 |  0.07 |  0.07 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.09 |  0.51 |  0.21 |  0.07 |  0.07 |  0.07 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Computing validation mAP (this may take a while)...

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  0.06 |  0.17 |  0.17 |  0.14 |  0.07 |  0.07 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
  mask |  0.14 |  0.59 |  0.59 |  0.07 |  0.07 |  0.07 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Tensorflow: 2.0.0
Pytorch: 1.6.0

Am I doing something wrong? For comparison purposes, I had used Darknet - YoloV4 to train which took less than a day for a slightly bigger size dataset.

Any help would be greatly appreciated.

Thanks

**Comment by GarudLimited** [comment]
I have same issue . can some one help

**Comment by maskrcnnuser** [comment]
Found the reason - I guess: COCO dataset and number of iterations is too large. Reduced the number of iterations and the ETA is come down.

**Comment by areebsyed** [comment]
> Found the reason - I guess: COCO dataset and number of iterations is too large. Reduced the number of iterations and the ETA is come down.

How did you reduce the number of iterations? did you change something in the train.py?

**Comment by maskrcnnuser** [comment]
> > Found the reason - I guess: COCO dataset and number of iterations is too large. Reduced the number of iterations and the ETA is come down.
> 
> How did you reduce the number of iterations? did you change something in the train.py?

Apologies for the delayed response. I changed config.py
